### DISCUSSION TITLE
extending a postProcessing filter

### DISCUSSION THREAD

**Discussion by wajda** [AbsaOSS/spline-spark-agent#722] [Q&A]
Hello, I am trying to use a postProcessing filter: 
![image](https://github.com/AbsaOSS/spline-spark-agent/assets/1311776/7113f795-ce87-4d3d-ad07-0035592bddde)

but I am getting this error:
Make sure that type NamedEntity is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
A full rebuild may help if 'LineageDispatcher.class' was compiled against an incompatible version of za.co.absa.commons.

I am using this version: spark-3.3-spline-agent-bundle_2.12. and spark 3.3.1
It was working before version 1.0. thanks a lot

_Originally posted by @kuhnen in https://github.com/AbsaOSS/spline-spark-agent/discussions/713#discussioncomment-6439398_

**Comment by wajda** [comment]
I'm not exactly sure what is happening. It might be caused by some mess in the classpath, but not 100% sure without having a deep look. Try to extend the `AbstractPostProcessingFilter` abstract class instead of `PostProcessingFilter` trait. Make sure you don't mixup old and new spine versions etc. Try a clean build of your project.

**Reply by kuhnen** [reply]
@wajda any class coming from commons is not visible when trying to compile with sbt. Already tried clean and rebuild. I also compiled the spline agent at my machine add the jar to lib/ and did not work :(

**Reply by kuhnen** [reply]
@wajda I believe it is related with this => 
![image](https://github.com/AbsaOSS/spline-spark-agent/assets/1311776/6ea7d0b7-d68e-42c3-a82d-66a489562426)

**Reply by kuhnen** [reply]
just tested with 1.2.0.  then it works.  I believe the issue is with the Shading

**Reply by wajda** [reply]
I see now. Yes, you are right. The `NamedEntity` trait accidentally appeared to be in the same package as the stuff from `absa-commons` library, and it happened to also be a part of the agent public API, hence the error. Will fix it asap.

**Reply by wajda** [reply]
Created a new issue #723. Closing this discussion.
